### PR TITLE
feat(Config): add rms assignment package templates data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add rms assignment package templates data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add rms assignment package templates data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rms/' TESTARGS='-run TestAccDatasourceTemplates_basic'             
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/rms/ -v -run TestAccDatasourceTemplates_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceTemplates_basic 
=== PAUSE TestAccDatasourceTemplates_basic 
=== CONT  TestAccDatasourceTemplates_basic 
--- PASS: TestAccDatasourceTemplates_basic (9.92s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rms       9.968s
```

### CODE REVIEW

File: docs/data-sources/rms_assignment_package_templates.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  The [parameters](#Templates_TemplateParameter) structure is documented below.
+
+<a name="Templates_TemplateParameter"></a>
+The `parameters` block supports:
+
+* `name` - Indicates the name of a parameter for a built-in assignment package template.
+
+* `description` - Indicates the description of a parameter for a built-in assignment package template.
+
+* `default_value` - Indicates the default value of a parameter for a built-in assignment package.
```
Comment (hmbig2) [inline docs/data-sources/rms_assignment_package_templates.md]:
```suggestion
* `default_value` - Indicates the default value of a parameter for a built-in assignment package template..
```

Comment (houpeng80) [reply docs/data-sources/rms_assignment_package_templates.md]: done

File: huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+	var templates []interface{}
+	var marker string
+	var getTemplatesPath string
+	for {
+		getTemplatesPath = getTemplatesBasePath + buildGetTemplatesQueryParams(d, marker)
+		getTemplatesResp, err := getTemplatesClient.Request("GET", getTemplatesPath, &getTemplatesOpt)
+
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving Identity Center groups")
```
Comment (ShiChangkuo) [inline huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go]: the error message should be updated

File: huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+			}
+			rst = append(rst, parameter)
+		}
+	}
+
+	return rst
+}
+
+func buildGetTemplatesQueryParams(d *schema.ResourceData, marker string) string {
+	res := "?limit=1"
```
Comment (ShiChangkuo) [inline huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go]: set limit to 200?

Comment (houpeng80) [reply huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go]: done

File: huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/rms/data_source_huaweicloud_rms_assignment_package_templates.go]: done

**Pull request merged**